### ISSUE TITLE
groupby_test.py sporadically fails

### ISSUE THREAD

**Issue by ronawho** [mhmerrill/arkouda#365]
test_groupby_on_one_level fails sporadically (1/20 runs) for me

```
======================================================= warnings summary ========================================================
tests/groupby_test.py::GroupByTest::test_groupby_on_one_level
  /Users/eronagha/Desktop/arkouda/tests/groupby_test.py:50: RuntimeWarning: invalid value encountered in subtract
    print(f"Different values (abs diff = {np.abs(pdvals - akvals).sum()})")

-- Docs: https://docs.pytest.org/en/latest/warnings.html
==================================================== short test summary info ====================================================
FAILED tests/groupby_test.py::GroupByTest::test_groupby_on_one_level - AssertionError: 0 != 1
============================================ 1 failed, 1 passed, 1 warning in 7.25s =============================================
```

**Comment by ronawho** [comment]
cc @mhmerrill, @reuster986, @hokiegeek2

**Comment by mhmerrill** [comment]
did we ever understand or track this down?

**Comment by hokiegeek2** [comment]
@mhmerrill No, but I would be happy to take it on once I wrap up #60

**Comment by mhmerrill** [comment]
I think we solved this with PR #434
